Sources/TuistKit/Services/TestService.swift

```diff
@@ -224,6 +229,28 @@ public final class TestService { // swiftlint:disable:this type_body_length
             )
         }
 
+        let passedResultBundlePath = resultBundlePath
+
+        let runResultBundlePath = try cacheDirectoryProviderFactory.cacheDirectories()
+            .tuistCacheDirectory(for: .runs)
+            .appending(components: runId, Constants.resultBundleName)
+
+        let resultBundlePath: AbsolutePath?
+        if config.cloud == nil {
+            resultBundlePath = passedResultBundlePath
+        } else {
+            resultBundlePath = passedResultBundlePath ?? runResultBundlePath
+        }
+
+        defer {
+            if let resultBundlePath, runResultBundlePath != resultBundlePath, config.cloud != nil {
+                if !FileHandler.shared.exists(runResultBundlePath.parentDirectory) {
+                    try? FileHandler.shared.createFolder(runResultBundlePath.parentDirectory)
+                }
+                try? FileHandler.shared.copy(from: resultBundlePath, to: runResultBundlePath)
+            }
+        }
```

Comment:
Are they in the magnitude of megabytes? I was thinking that instead of building something specific fo result bundles, that we can have a more generic clean logic that runs periodically to store artifacts that are older than date X. If you agree with this approach, I'd then remove this logic, and implement that separately.

Reply:
They are in order of megabytes but they get automatically removed as soon as the result bundle gets uploaded to the server.